File: sort_algorithms/insertion_sort/python/insertion_sort.py
def insertion_sort(arr):
    """
    Sorts an array of elements in ascending order using the insertion sort algorithm.

    Parameters:
    arr (list): The list of elements to be sorted.

    Returns:
    list: The sorted list of elements.
    """
    for i in range(1, len(arr)):
        current_element = arr[i];
        j = i - 1;
        while j >= 0 and current_element < arr[j]:
            arr[j + 1] = arr[j];
            j -= 1;
        arr[j + 1] = current_element;
    return arr

not_sort_array = [5, 3, 8, 4, 2];

insertion_sort(not_sort_array)

File: sort_algorithms/insertion_sort/javascript/insertion_sort.js
/**
 * Sorts an array using the insertion sort algorithm.
 *
 * @param {number[]} sortedList - The array to be sorted.
 * @returns {number[]} The sorted array.
 */
const nonSortedArray = [5, 3, 8, 4, 2]
const inputList = [...nonSortedArray]


function insertSort(sortedList) {
    for (let i = 1; i < (sortedList.length); i += 1) {
        const currentElement = sortedList[i];
        let j = i - 1;
        while (j >= 0 && sortedList[j] > currentElement) {
            sortedList[j + 1] = sortedList[j];
            j -= 1; 
        }
        sortedList[j + 1] = currentElement;
    }
  return sortedList;
}

console.log('not sorted array: ', nonSortedArray)
console.log('sorted array: ', insertSort(inputList))


const outputElement = document.getElementById('output');
outputElement.innerHTML = '';
outputElement.innerHTML = `
  <p>Not sorted array: ${nonSortedArray}</p>
  <p>Sorted array: ${inputList}</p>
  <p>For details, check the console.</p>
`;